src/MagnumPlugins/BasisImageConverter/Test/BasisImageConverterTest.cpp

```diff
@@ -0,0 +1,276 @@
+    #ifdef PNGIMPORTER_PLUGIN_FILENAME
+    CORRADE_INTERNAL_ASSERT(_importerManager.load(PNGIMPORTER_PLUGIN_FILENAME) & PluginManager::LoadState::Loaded);
+    #endif
+    /* Load the plugin directly from the build tree. Otherwise it's static and
+       already loaded. */
+    #ifdef BASISIMAGECONVERTER_PLUGIN_FILENAME
+    CORRADE_INTERNAL_ASSERT(_converterManager.load(BASISIMAGECONVERTER_PLUGIN_FILENAME) & PluginManager::LoadState::Loaded);
+    #endif
+    #ifdef BASISIMPORTER_PLUGIN_FILENAME
+    CORRADE_INTERNAL_ASSERT(_importerManager.load(BASISIMPORTER_PLUGIN_FILENAME) & PluginManager::LoadState::Loaded);
+    #endif
+}
+
+void BasisImageConverterTest::wrongFormat() {
+    Containers::Pointer<AbstractImageConverter> converter =
+        _converterManager.instantiate("BasisImageConverter");
+    ImageView2D image{PixelFormat::RG32F, {}, nullptr};
+
+    std::ostringstream out;
+    Error redirectError{&out};
+    CORRADE_VERIFY(!converter->exportToData(image));
+    CORRADE_COMPARE(out.str(), "Trade::BasisImageConverter::exportToData(): unsupported format PixelFormat::RG32F\n");
+}
+
+void BasisImageConverterTest::zeroSize() {
+    Containers::Pointer<AbstractImageConverter> converter =
+        _converterManager.instantiate("BasisImageConverter");
+
+    std::ostringstream out;
+    Error redirectError{&out};
+    CORRADE_VERIFY(!converter->exportToData(ImageView2D{PixelFormat::RGB8Unorm, {}, nullptr}));
+    CORRADE_COMPARE(out.str(),
+        "Trade::BasisImageConverter::exportToData(): source image is empty\n");
+}
+
+void BasisImageConverterTest::r() {
+    if(_importerManager.loadState("PngImporter") == PluginManager::LoadState::NotFound)
+        CORRADE_SKIP("PngImporter plugin not found, cannot test contents");
+
+    Containers::Pointer<AbstractImporter> pngImporter =
+        _importerManager.instantiate("PngImporter");
+    pngImporter->openFile(Utility::Directory::join(BASISIMPORTER_TEST_DIR, "rgb-27x63.png"));
+    const auto originalImage = pngImporter->image2D(0);
+
+    /* Create image with skip and padding for aligning rows to 4 bytes */
+    const Containers::Array<UnsignedByte> imageDataWithSkip{Containers::ValueInit,
+        ((27+8) + 1)*(63+7)};
+    const ImageView2D imageWithSkip{PixelStorage{}.setSkip({0, 1, 0}),
+        PixelFormat::R8Unorm, originalImage->size(), imageDataWithSkip};
+
+    auto sourcePixels = originalImage->pixels<Color4ub>();
+    auto destPixels = imageWithSkip.pixels<UnsignedByte>();
+    for(std::size_t y = 0; y != sourcePixels.size()[0]; ++y)
+        for(std::size_t x = 0; x != sourcePixels.size()[1]; ++x)
+            destPixels[y][x] = sourcePixels[y][x].r();
+
+    const auto compressedData = _converterManager.instantiate("BasisImageConverter")->exportToData(imageWithSkip);
+    CORRADE_VERIFY(compressedData);
+
+    if(_importerManager.loadState("BasisImporterRGBA8") == PluginManager::LoadState::NotFound)
+        CORRADE_SKIP("BasisImporterRGBA8 plugin not found, cannot test");
```

Comment:
Just check for `BasisImporter` alone, this error message may be confusing for people who don't know the aliases. Same below.

Reply:
But I guess I would check for `BasisImporterRGBA8`, but just change the error message, right?

Reply:
If there's `BasisImporter` and it doesn't provide a `BasisImporterRGBA8` alias, then there's something very off and it should blow up (while this would lead to just `SKIP`) ... so pick the loudest alternative :D

But this is a minor thing, nothing critical.

Reply:
Well, not necessarily, could just be compiled out or an older version 🤷‍♂ 

Not sure what your comment means in practice, shall I remove the RGBA8 prefix from both the skip message and the loadState call?

Reply:
Yes, remove both.